### PR TITLE
Add unit test for `get_input_variables`

### LINKED ISSUE #23: Add unit test: utils: `get_input_variables`
The function `get_input_variables` does not have a unit test. It should have one.

### PR DESCRIPTION
**Pull request by mary-cleaton**
## Description

Use mocking to add a new unit test for `get_input_variables`, since this calls seven child functions which each already have a unit test. Whilst doing this, also tweaked `read_configs` so that `get_input_variables` only has to call it once and updated `test_read_configs` to reflect this change.

closes #23

## Type of change

*You can delete options that are not relevant.*

- [ ] Bug fix - *non-breaking change*
- [x] New feature - *non-breaking change*
- [ ] Breaking change - *backwards incompatible change, changes expected behaviour*
- [x] Non-user facing change, structural change, dev functionality, docs ...

## Checklist:

- [x] I have performed a self-review of my own code.
- [x] I have commented my code appropriately, focusing on explaining my design decisions (explain why, not how).
- [x] I have made corresponding changes to the documentation (comments, docstring, etc.. )
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have updated the change log.

<br>

#  Peer review
Any new code includes all the following:

- **Documentation**: docstrings, comments have been added/ updated.
- **Style guidelines**: New code conforms to the project's contribution guidelines.
- **Functionality**: The code works as expected, handles expected edge cases and exceptions are handled appropriately.
- **Complexity**: The code is not overly complex, logic has been split into appropriately sized functions, etc..
- **Test coverage**: Unit tests cover essential functions for a reasonable range of inputs and conditions. Added and existing tests pass on my machine.

### Review comments
Suggestions should be tailored to the code that you are reviewing. Provide context.
Be critical and clear, but not mean. Ask questions and set actions.
<details><summary>These might include:</summary>

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented
  - Do the tests effectively assure that it
  works correctly? Are there additional edge cases/ negative tests to be considered?
- code style improvements (could the code be written more clearly?)
</details>
<br>

*Further reading: [code review best practices](https://best-practice-and-impact.github.io/qa-of-code-guidance/peer_review.html)*

### CODE REVIEW

File: tests/utils/test_utils.py [hunk 1]
```diff
@@ -275,12 +359,10 @@ def test_read_configs():
...
 
     # Act
-    test_output = ut.read_configs(
-        config_path=test_config_path, config_section="section_1"
-    )
+    test_output = ut.read_configs(config_path=test_config_path)
 
     # Assert
-    assert test_output["variables"] == "a, b, c, d"
+    assert test_output["section_1"]["variables"] == "a, b, c, d"
```
Comment (alani-odunlami) [inline tests/utils/test_utils.py:365]: Wheree does  'section_1' in assert test_output["section_1"]["variables"]  comes from?

Comment (mary-cleaton) [reply tests/utils/test_utils.py:365]: It's from lines 356-357.

Comment (alani-odunlami) [reply tests/utils/test_utils.py:365]: Okay, thanks.

**Review by alani-odunlami — APPROVED** [review]
Code is sufficiently documented which aids my understanding. E.g,  the function test_get_input_variables contains docstrings within it. 

So, I'm happy with the updates.
